examples/sparse_blas/compile_time_dispatching/sparse_blas_gemv_usm_mklcpu.cpp

```diff
@@ -0,0 +1,249 @@
+
+    ia = (intType *)sycl::malloc_shared(sizeia * sizeof(intType), cpu_queue);
+    ja = (intType *)sycl::malloc_shared(sizeja * sizeof(intType), cpu_queue);
+    a = (fp *)sycl::malloc_shared(sizea * sizeof(fp), cpu_queue);
+    x = (fp *)sycl::malloc_shared(sizevec * sizeof(fp), cpu_queue);
+    y = (fp *)sycl::malloc_shared(sizevec * sizeof(fp), cpu_queue);
+    z = (fp *)sycl::malloc_shared(sizevec * sizeof(fp), cpu_queue);
+
+    if (!ia || !ja || !a || !x || !y || !z) {
+        throw std::runtime_error("Failed to allocate USM memory");
+    }
+
+    intType nnz = generate_sparse_matrix<fp, intType>(size, ia, ja, a);
+
+    // Init vectors x and y
+    for (int i = 0; i < nrows; i++) {
+        x[i] = set_fp_value(fp(1.0));
+        y[i] = set_fp_value(fp(0.0));
+        z[i] = set_fp_value(fp(0.0));
+    }
+
+    std::vector<intType *> int_ptr_vec;
+    int_ptr_vec.push_back(ia);
+    int_ptr_vec.push_back(ja);
+    std::vector<fp *> fp_ptr_vec;
+    fp_ptr_vec.push_back(a);
+    fp_ptr_vec.push_back(x);
+    fp_ptr_vec.push_back(y);
+    fp_ptr_vec.push_back(z);
+
+    //
+    // Execute Matrix Multiply
+    //
+
+    oneapi::mkl::transpose transA = oneapi::mkl::transpose::nontrans;
+    std::cout << "\n\t\tsparse::gemv parameters:\n";
+    std::cout << "\t\t\ttransA = "
+              << (transA == oneapi::mkl::transpose::nontrans
+                      ? "nontrans"
+                      : (transA == oneapi::mkl::transpose::trans ? "trans" : "conjtrans"))
+              << std::endl;
+    std::cout << "\t\t\tnrows = " << nrows << std::endl;
+    std::cout << "\t\t\talpha = " << alpha << ", beta = " << beta << std::endl;
+
+    // create and initialize handle for a Sparse Matrix in CSR format
+    oneapi::mkl::sparse::matrix_handle_t handle = nullptr;
+
+    oneapi::mkl::sparse::init_matrix_handle(cpu_selector, &handle);
+
+    auto ev_set = oneapi::mkl::sparse::set_csr_data(cpu_selector, handle, nrows, nrows, nnz,
+                                                    oneapi::mkl::index_base::zero, ia, ja, a);
+
+    auto ev_opt = oneapi::mkl::sparse::optimize_gemv(cpu_selector, transA, handle, { ev_set });
+
+    auto ev_gemv =
+        oneapi::mkl::sparse::gemv(cpu_selector, transA, alpha, handle, x, beta, y, { ev_opt });
+
+    auto ev_release =
+        oneapi::mkl::sparse::release_matrix_handle(cpu_selector, &handle, { ev_gemv });
+
+    ev_release.wait_and_throw();
+
+    //
+    // Post Processing
+    //
+
+    fp *res = y;
+    for (intType row = 0; row < nrows; row++) {
+        fp tmp = set_fp_value(fp(0.0));
+        for (intType i = ia[row]; i < ia[row + 1]; i++) {
+            tmp += a[i] * x[ja[i]];
+        }
+        z[row] = alpha * tmp + beta * z[row];
+    }
+
+    fp diff = set_fp_value(fp(0.0));
+    for (intType i = 0; i < nrows; i++) {
+        diff += (z[i] - res[i]) * (z[i] - res[i]);
+    }
+    std::cout << "\n\t\t sparse::gemv residual:\n"
+              << "\t\t\t" << diff << "\n\tFinished" << std::endl;
```

Comment:
Please change the validation approach here to match the oneMKL 2024.0 gold release. Checking `diff > 0` a few lines below does not work for `std::complex<>` data type, so users will get build errors if they try to simply change from float to complex data types manually in this example.